src/include/sof/common.h

```diff
@@ -177,6 +177,33 @@
 #define IS_ENABLED_STEP_3(ignore, value, ...) (!!(value))
 #endif
 
+#define SOF_CONFIG_HIFI(level, component) (CONFIG_ ## component ## _HIFI_ ## level)
+
+/* True if:
+ *  (1) EITHER this particular level was manually forced in Kconfig,
+ *  (2) OR:  - this component defaulted to "MAX"
+ *           - AND this level is the max available in the XC HAL.
+ */
+#define SOF_USE_HIFI(level, component) (SOF_CONFIG_HIFI(level, component) || \
+	(SOF_CONFIG_HIFI(MAX, component) && level == SOF_MAX_XCHAL_HIFI))
+
+#ifndef __XCC__ // Cadence toolchains: either xt-xcc or xt-clang.
```

Comment:
those will eventually be removed, I am fully understand you have 100% confidence on these definitions, however, it is still have possibility to have unexpected situations to handling, if introduce new bugs, then what we got(less code duplicate) is less than what we lost, just my opinion.